library/Mockery/Generator/Parameter.php

```diff
@@ -56,7 +62,13 @@ public function getTypeHintAsString()
         }
 
         if (preg_match('/^Parameter #[0-9]+ \[ \<(required|optional)\> (?<typehint>\S+ )?.*\$' . $this->rfp->getName() . ' .*\]$/', $this->rfp->__toString(), $typehintMatch)) {
+
             if (!empty($typehintMatch['typehint'])) {
+
+                if ($this->isNamespacedClass($typehintMatch['typehint'])) {
```

Comment:
Currently the only ofther thing could be interface name or `array`/`callable` word. If the scalar type typehints RFC will be accepted, then there could be more.

I don't know any method that can tell that given string is one of build-in types.

Maybe also typehints can be `self` or `static`, but I'm not sure at 100%.